api/src/main/java/io/opentelemetry/api/baggage/BaggageBuilder.java

```diff
@@ -51,7 +51,7 @@
    * @param entryMetadata the {@code EntryMetadata} associated with this {@link Entry}.
    * @return this
    */
-  BaggageBuilder put(String key, String value, EntryMetadata entryMetadata);
+  BaggageBuilder put(String key, String value, BaggageEntryMetadata entryMetadata);
```

Comment:
API compatible but ABI incompatible. I could add an overload for the latter too, but guess it shouldn't matter at this stage.